### ISSUE TITLE
Task arguments not parsing correctly

### ISSUE THREAD

**Issue by dgeswein-dlx** [spring-cloud/spring-cloud-dataflow#4782]
I am having difficulty passing a JSON string as an argument to a dataflow task.  The quotes and spaces within the JSON are problematic for the parser.  I have narrowed the issue to the following method within the dataflow server:  org.springframework.cloud.dataflow.rest.util.DeploymentPropertiesUtils.parseArgumentList(String s, String delimiter)

It appears that I can surround my string with quotes to ensure it stays intact, however there is not a clear way to escape the quotes within (don't see anything in the parser that allows for this).

Interestingly, it will parse JSON with spaces and quotes, but only if it is in an odd position within the argument list.

This argument list parses correctly:
`arg2="Argument 2" arg3=val3`

This argument list does not (the second argument does not remain intact):
`arg1=val1 arg2="Argument 2" arg3=val3`

This argument list parses correctly:
`arg0=val0 arg1=val1 arg2="Argument 2" arg3=val3`

I am seeing multiple issues here:

1) double quoting of arguments with spaces does not preserve the argument value unless it is in an odd position
2) There does not appear to be a way to escape double quotes within the double quoted text (necessary for JSON)

**Comment by aritzbastida** [comment]
I'm having the same issue. As I can see in the logs, the double quotes are escaped by Local Task Launcher (did not try with other deployers yet):

`2022-02-07 12:13:58.330  INFO 18336 --- [nio-9393-exec-4] o.s.c.d.spi.local.LocalTaskLauncher      : Command to be executed: java.exe -jar sample-task.jar arg=value base.dir.name=\"D:/TEMP/DIR WITH SPACES\" arg2=value --spring.cloud.task.executionid=4`

As a result, job parameters are misinterpreted by Spring Batch.

**Comment by dgeswein-dlx** [comment]
When I opened this issue, I was using the Kubernetes Deployer.  The above report from @aritzbastida looks like slightly different behavior than I was experiencing and may even be a different issue related to arg parsing.  In any case, parsing of arguments could be improved in DeploymentPropertiesUtils and perhaps other layers as well.

**Comment by bjafet123** [comment]
Hi all

I have this issue and I think is related but don't know how to solve it. I'm using the spring-cloud-dataflow-server-2.9.4. I'm trying to obtain a JSON, the issue is when the stream is redeployed.

When I configure my stream to be depoyed, in which I'm using a processor (transform, script or http-request), in the "expression" atribute I need to set an expression that contains quotes and double quotes (escaped). The expression works properly the first time I set and allows to deploy the stream, but if I undeploy the stream and try again to deploy it, the spring cloud data flow throws state machine exception because the backslashes used to escape the double quotes are removed.

I already follow the considerations in the Spaces and Quotes documentation, but I think that it only applies to the streams definition and not to the deployment time.

The URL of the spaces and quotes documentarion is: https://docs.spring.io/spring-cloud-dataflow/docs/current-SNAPSHOT/reference/htmlsingle/#shell-white-space

And the sample of the type of expression required:

`expression="new String('{\"size\": 1,\"sort\": {\"timestamp\": \"desc\"},\"query\": {\"prefix\": {\"integrationname\": \"63320e0d3134667225f\"}}}')"`

The stream could be as simple as:

`http | transform | log`

The firs time the expression is set looks like as follows:

`expression="new String('{\"size\": 1,\"sort\": {\"timestamp\": \"desc\"},\"query\": {\"prefix\": {\"integrationname\": \"63320e0d313862934667225f\"}}}')"`

Deploying correctly the stream.

Once the stream is undeployed and try to deploy it again, the espression looks like:

`expression="new String('{"size": 1,"sort\": {"timestamp": "desc"},"query": {"prefix": {"integrationname": "63320e0d313862934667225f"}}}')"`

Where the backslashes were removed, causing the state machine exception because of the unescaped double quotes.

Thanks in advance
